diff --git a/module/evdi_fb.c b/module/evdi_fb.c
index 60895b2..d8442e8 100644
--- a/module/evdi_fb.c
+++ b/module/evdi_fb.c
@@ -213,8 +213,11 @@ static const struct fb_ops evdifb_ops = {
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
+#if KERNEL_VERSION(7, 0, 0) <= LINUX_VERSION_CODE
+#else
 	.fb_debug_enter = drm_fb_helper_debug_enter,
 	.fb_debug_leave = drm_fb_helper_debug_leave,
+#endif
 	.fb_mmap = evdi_fb_mmap,
 	.fb_open = evdi_fb_open,
 	.fb_release = evdi_fb_release,
